### ISSUE TITLE
how can I import a backup? how can I change my password ?

### ISSUE THREAD

**Issue by longbill** [BitcoinMafia/SpareCoinsExt#17]
how can I import a backup? how can I change my password ?

**Comment by longbill** [comment]
this extension is awesome. thank you. I love it.

**Comment by scottlingran** [comment]
Hey, thanks for the love! 

We haven't built the import feature yet. Wanna make a pull request? :)

**Comment by longbill** [comment]
yes. I am planning to build this feature. But I have no idea now. Can you give me some rough idea about how to do that ?  

---

sent from mac

在 2014年7月5日 星期六，下午6:29，Scott Li 写道：

**Comment by scottlingran** [comment]
How you done Chrome Extensions before? The official tutorial is actually pretty good - https://developer.chrome.com/extensions/getstarted

The app is built with angular.

You should be able to pull the repo and do
- Window > Extensions
- Enable "developer mode"
- Load unpacked extension (from cloned repo)

**Comment by longbill** [comment]
Hi, I am a professional developer. I know how to create an extension and angular. Could you give me some idea about how to build the import backup and change password feature?

---

sent from mac

在 2014年7月9日 星期三，上午11:00，Scott Li 写道：

**Comment by scottlingran** [comment]
The SHA256 digest of the user's passwords is stored in ChromeStorage

https://github.com/BitcoinMafia/SpareCoinsExt/blob/master/js/controllers/password.js#L32-34

I imagine you'd need to 
1. Confirm old password first using the digest from user password input and the digest in ChromeStorage
2. If it matches, then allow user to set/confirm a new password
3. The old digest should be overriden or removed

To increase security, it's best to salt it as well.

The encryption of the private keys when backing up is here. You should the SHA256 digest of your password to decrypt.

https://github.com/BitcoinMafia/SpareCoinsExt/blob/master/background.js

Hope that helps

**Comment by longbill** [comment]
Thank you for your help. I will let you know if I got any progress.  Do you have a schedule to build these features by yourself ?

---

sent from mac

在 2014年7月10日 星期四，上午11:04，Scott Li 写道：

**Comment by scottlingran** [comment]
Nope, it's all yours
